gridfs/asynchronous/grid_file.py

```diff
@@ -1021,6 +1021,36 @@ async def rename(
                 "matched file_id %i" % (new_filename, file_id)
             )
 
+    async def rename_by_name(
+        self, filename: str, new_filename: str, session: Optional[AsyncClientSession] = None
+    ) -> None:
+        """Renames the stored file with the specified filename.
+
+        For example::
+
+          my_db = AsyncMongoClient().test
+          fs = AsyncGridFSBucket(my_db)
+          await fs.upload_from_stream("test_file", "data I want to store!")
+          await fs.rename_by_name("test_file", "new_test_name")
+
+        Raises :exc:`~gridfs.errors.NoFile` if no file with file_id exists.
+
+        :param filename: The filename of the file to be renamed.
+        :param new_filename: The new name of the file.
+        :param session: a
+            :class:`~pymongo.client_session.AsyncClientSession`
+
+        .. versionadded:: 4.12
+        """
+        _disallow_transactions(session)
+        result = await self._files.update_many(
+            {"filename": filename}, {"$set": {"filename": new_filename}}, session=session
+        )
+        if not result.matched_count:
+            raise NoFile(
+                f"no files could be renamed {new_filename} because none matched filename {filename}"
```

Comment:
Suggest: `f"no files could be renamed {new_filename!r} because none matched filename {filename!r}"` to make the strings more obvious.